lib/AST/ASTPrinter.cpp

```diff
@@ -2011,6 +2026,27 @@ void PrintAST::visitExtensionDecl(ExtensionDecl *decl) {
     printExtension(decl);
 }
 
+/// Determines if we should print the text of the initializer expression.
+/// There's a very narrow case when we would:
+/// If the decl:
+///   - has an initializer expression
+///   - is @usableFromInline
+/// If the parent type:
+///   - is @_fixed_layout (or effectively fixed-layout)
+///   - has at least one inlinable initializer in itself or any of its
+///     extensions.
+static bool shouldPrintInitializerExpr(const VarDecl *vd) {
+  if (!vd->getAttrs().hasAttribute<HasInitialValueAttr>() &&
+      !vd->getAttrs().hasAttribute<UsableFromInlineAttr>())
+    return false;
+  auto parent = dyn_cast<NominalTypeDecl>(vd->getDeclContext());
+  if (!parent) return false;
+  if (parent->isFormallyResilient())
+    return false;
+  return evaluateOrDefault(vd->getASTContext().evaluator,
```

Comment:
I don't understand why this needs to be a request. In fact I don't think you need to check this condition at all.

Reply:
If we are doing the check, it’s nice not to look through the members and extensions for each property of the type. Though, if we don’t need the check all of the request plumbing can go away.